dlr/dlr-engine/src/main/java/ai/djl/dlr/jni/LibUtils.java

```diff
@@ -24,11 +44,302 @@
 @SuppressWarnings("MissingJavadocMethod")
 public final class LibUtils {
 
+    private static final Logger logger = LoggerFactory.getLogger(LibUtils.class);
+
+    private static final String LIB_NAME = "djl_dlr";
+    private static final String NATIVE_LIB_NAME = "dlr";
+
+    private static final Pattern VERSION_PATTERN =
+            Pattern.compile("(\\d+\\.\\d+\\.\\d+(-\\w)?)(-SNAPSHOT)?(-\\d+)?");
+
     private LibUtils() {}
 
     public static void loadLibrary() {
-        // TODO implement
-        System.load("/Users/leecheng/workspace/djl/dlr/dlr-native/libdlr.dylib");
-        System.load("/Users/leecheng/.djl.ai/dlr/1.5.0-cpu-osx-x86_64/libdjl_dlr.dylib");
+        String nativePath = findNativeOverrideLibrary();
+        Path cacheDir;
+        AtomicBoolean fallback = new AtomicBoolean(false);
+        if (nativePath == null) {
+            String nativeLibDir = findNativeLibrary(fallback);
+            if (nativeLibDir != null) {
+                cacheDir = Paths.get(nativeLibDir);
+                nativePath = nativeLibDir + "/" + System.mapLibraryName(NATIVE_LIB_NAME);
+            } else {
+                throw new IllegalStateException("Native library not found");
+            }
+        } else {
+            Platform platform;
+            try {
+                platform = getPlatformFromProperties();
+            } catch (IOException e) {
+                throw new IllegalStateException(
+                        "Failed to read DLR native library jar properties", e);
+            }
+            if (platform == null) {
+                throw new IllegalStateException(
+                        "Your DLR native library jar does not match your operating system. Make sure the Maven Dependency Classifier matches your system type.");
+            }
+            cacheDir = getCacheDir(platform);
+        }
+        String jniPath = copyJniLibraryFromClasspath(cacheDir, fallback.get());
```

Comment:
DLR jni should be work for both CPU/GPU build right? then we don't need fallback.

Reply:
right! removed it